iothub/device/tests/Authentication/EdgeDeviceOnBehalfOfTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text;
+using System.Threading.Tasks;
+using System;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client.Authentication;
+
+namespace Microsoft.Azure.Devices.Client.Tests.OnBehalfOf
+{
+    public class EdgeDeviceOnBehalfOfTests
+    {
+        private static string _testKey => Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.Empty.ToString("N")));
+
+        [TestMethod]
+        public async Task ConnectDeviceOnBehalfOf()
+        {
+            string leafDeviceId = "test-leaf-device";
+
+            DeviceConnectionSettings edgeDevice = new DeviceConnectionSettings("e4k-hub.azure-devices.net", "test-edge-device")
```

Comment:
`DeviceConnectionSettings` and `EdgeHubConnectionSettingsBuilder` are classes that don't exist yet -- I'm looking for guidance on whether to add them from the e4k connector codebase or to find analogous places here in V2 to replace their functionality for these tests

Reply:
I would not include the DeviceConnectionSettings in V2. The class is used just to hold the connection settings, you should be able to use IotHubConnectionString or just the individual values required to establish the connection